src/back/External/Api/Actions/TopicsDetails.php

```diff
@@ -26,24 +26,35 @@ trait TopicsDetails
      */
     public function getTopicsDetails(
         array           $topics,
-        TopicSearchMode $searchMode = TopicSearchMode::ID
+        TopicSearchMode $searchMode,
     ): ApiError|TopicsDetailsResponse {
         /** @var TopicDetails[] $knownTopics */
         $knownTopics = [];
 
         $client = $this->client;
 
-        $chunkErrorHandler = self::getChunkErrorHandler($this->logger);
-        $topicProcessor    = self::getTopicDetailsProcessor($this->logger, $knownTopics);
+        $chunkErrorHandler   = self::getChunkErrorHandler($this->logger);
+        $topicProcessor      = self::getTopicDetailsProcessor($this->logger, $knownTopics);
+        $minRequestThreshold = $this->minRequestThreshold;
 
         /**
          * @param array[][] $optionsChunks
          *
          * @return Generator
          */
-        $requests = function(array $optionsChunks) use (&$client) {
+        $requests = function(array $optionsChunks) use (&$client, $minRequestThreshold) {
+            $lastRequestTime = 0;
             foreach ($optionsChunks as $chunk) {
-                yield function(array $options) use (&$client, &$chunk) {
+                yield function(array $options) use (&$client, &$chunk, &$lastRequestTime, $minRequestThreshold) {
+                    if ($minRequestThreshold > 0) {
+                        $now     = microtime(true);
+                        $elapsed = $now - $lastRequestTime;
+                        if ($elapsed < $minRequestThreshold) {
+                            usleep((int) (($minRequestThreshold - $elapsed) * 1_000_000));
+                        }
+                        $lastRequestTime = microtime(true);
+                    }
+
```

Comment:
503 code processing is done in getRetryMiddleware. 
If the API responds with an error, the request will be repeated later automatically. 

Have you compared the update time of a large number of topics in its current form and with the fixes made? What is the point of them?

https://github.com/keepers-team/webtlo/blob/master/src/back/External/Shared/RetryMiddleware.php#L27-L62
https://github.com/caseyamcl/guzzle_retry_middleware?tab=readme-ov-file#usage

Reply:
Also, `Pool` used `concurrency=4`. In my testing I got 503 0 times.

If you want to see some logs, just add here
https://github.com/keepers-team/webtlo/blob/master/src/back/External/Api/Actions/TopicsDetails.php#L110
line like this
```php
$logger->debug('Done chunk request', ['index' => $index]);
```

Reply:
The point of this change is to avoid the 503 code at all - we should not make requests to the API with the frequency higher than the rate limit. I was getting numerous 503 errors before I implemented this change.
[503.zip](https://github.com/user-attachments/files/20166037/503.zip)

Reply:
Just change concurrency to 2.